lib/cli/run.js

```diff
@@ -0,0 +1,392 @@
+'use strict';
+
+/**
+ * Command module for default ("run") command
+ */
+
+const path = require('path');
+const {loadConfig, findConfig} = require('./config');
+const Mocha = require('../mocha');
+const chalk = require('chalk');
+
+const {
+  list,
+  cwd,
+  showInterfaces,
+  showReporters,
+  handleFiles,
+  handleRequires,
+  validatePlugin,
+  runMocha
+} = require('./run-helpers');
+const debug = require('debug')('mocha:cli:run');
+
+/**
+ * Logical option groups
+ */
+const GROUPS = {
+  FILES: 'File Handling',
+  FILTERS: 'Test Filters',
+  NODEJS: 'Node.js & V8',
+  OUTPUT: 'Reporting & Output',
+  RULES: 'Rules & Behavior',
+  CONFIG: 'Config Files'
+};
+
+/**
+ * Placeholder value to force config parsing function to run.
+ */
+const CONFIG_PATH_PLACEHOLDER = '__MOCHA_CONFIG_PATH__';
+
+const SHOW_HIDDEN_HELP_OPTION = (exports.SHOW_HIDDEN_HELP_OPTION =
+  'show-all-options');
+
+let cachedConfig;
+
+exports.command = '$0 [spec..]';
+
+exports.describe = 'Run tests with Mocha';
+
+exports.builder = yargs =>
+  yargs
+    .showHidden(
+      SHOW_HIDDEN_HELP_OPTION,
+      'Show help, including all Node.js/V8 & other hidden options'
+    )
+    .options({
+      'allow-uncaught': {
+        description: 'Allow uncaught errors to propagate',
+        group: GROUPS.RULES,
+        type: 'boolean'
+      },
+      'async-only': {
+        alias: 'A',
+        description:
+          'Require all tests to use a callback (async) or return a Promise',
+        group: GROUPS.RULES,
+        type: 'boolean'
+      },
+      'check-leaks': {
+        description: 'Check for global variable leaks',
+        group: GROUPS.RULES,
+        type: 'boolean'
+      },
+      'forbid-only': {
+        description: 'Fail if exclusive test(s) encountered',
+        group: GROUPS.RULES,
+        type: 'boolean'
+      },
+      'forbid-pending': {
+        description: 'Fail if pending test(s) encountered',
+        group: GROUPS.RULES,
+        type: 'boolean'
+      },
+      'full-trace': {
+        description: 'Display full stack traces',
+        group: GROUPS.OUTPUT,
+        type: 'boolean'
+      },
+      'inline-diffs': {
+        description:
+          'Display actual/expected differences inline within each string',
+        group: GROUPS.OUTPUT,
+        type: 'boolean'
+      },
+      'no-colors': {
+        alias: 'C',
+        conflicts: ['colors'],
+        description: 'Disable color output',
+        group: GROUPS.OUTPUT,
+        type: 'boolean'
+      },
+      'no-timeouts': {
+        conflicts: ['timeout'],
+        description: 'Disable timeouts',
+        group: GROUPS.RULES,
+        type: 'boolean'
+      },
```

Comment:
Had no problems with either of the two above and the default handling of options beginning with `--no-` (`yargs-parser` default settings)?

Reply:
hm, the only reason this was here was because it was in the help output before.

it'd probably be better to remove stuff like this and write something about how all boolean options can be prefixed with `--no-` to disable them.

- [ ] do that.